### PR TITLE
resource/gitlab_service_jira: several cleanup of tech debt

### PR DESCRIPTION
**Pull request by timofurrer**
This change set cleans up a few things in the `gitlab_service_jira` resource:

* properly use resource ID in read function
* use passthrough importer
* state in docs that `jira_issue_transition_id` attribute cannot be imported
* add import verification test steps to `_basic` test and remove dedicated `_import` test

### CODE REVIEW

File: internal/provider/resource_gitlab_service_jira.go
Comment (PatrickRice-KSC) [inline internal/provider/resource_gitlab_service_jira.go:190]: @timofurrer  - I assume we'll want to create an issue for this FIXME when we merge this so that we can investigate which of these two items it is 😄 .

I have a docker compose (and a jira license) that I can use to test this out, so I can take a look at it later.

**Review by PatrickRice-KSC — APPROVED** [review]
Overall, LGTM! 🎉

File: internal/provider/resource_gitlab_service_jira.go
Comment (timofurrer) [reply internal/provider/resource_gitlab_service_jira.go:190]: @PatrickRice-KSC That would be great. 

However, I'm not sure how Jira would be involved here - I think it's a GitLab / go-gitlab issue here only, but please give it a try.
IMHO: we can also wait for that until a merge to hopefully avoid the FIXME entirely ...

Comment (timofurrer) [reply internal/provider/resource_gitlab_service_jira.go:190]: @PatrickRice-KSC I didn't yet have time to dig into it, but just spotted this fishy part when unmarshalling the JSON from the API and fiddling with that very field: 

https://github.com/xanzy/go-gitlab/blob/65486a91e11bbb63cbf61bce23cfb48ee174b4d5/services.go#L673

Comment (PatrickRice-KSC) [reply internal/provider/resource_gitlab_service_jira.go:190]: @timofurrer  Yeah, that definitely seems a bit fishy. I did just try this out, and attaching a JIRA instance doesn't make any difference, it still doesn't unmarshall properly, so we can rule that out.

**Review by armsnyder — DISMISSED** [review]
🥳

File: internal/provider/resource_gitlab_service_jira.go
Comment (timofurrer) [reply internal/provider/resource_gitlab_service_jira.go:190]: I've created an upstream issue here: https://gitlab.com/gitlab-org/gitlab/-/issues/362437

At the moment I think we can merge this. WDYT? @PatrickRice-KSC

**Review by armsnyder — APPROVED** [review]

File: internal/provider/resource_gitlab_service_jira.go
Comment (PatrickRice-KSC) [reply internal/provider/resource_gitlab_service_jira.go:190]: Agreed!

**Review by PatrickRice-KSC — APPROVED** [review]

**Pull request merged**
